### PR TITLE
ABC Lesson

### PR DESCRIPTION
**Pull request by surabhigupta2029**
Creates the ABC lesson! 
Consists of alphabet audio, example words and images

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,4 +48,4 @@
       "last 1 safari version"
     ]
   }
-}
+}
```
Comment (ThaiQ) [inline package.json]: empty line

File: src/components/lesson-practice/practice.css [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    color: rgb(5, 0, 0);
+    font-weight: 500;
+    text-align: center;
+    cursor: pointer;
+    margin-left: 10px;
+    margin-right: 10px;
+    padding-left: 10%;
+    padding-right: 10%;
+    width: 250px;
+}
```
Comment (ThaiQ) [inline src/components/lesson-practice/practice.css]: empty lines

File: src/pages/ABCLesson/ABCLesson.css [hunk 3]
```diff
@@ -0,0 +1,176 @@
+@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@300&display=swap');
+
+.game{
+    color: #ddd;
+    background-color: #9c6176;
+    padding-top: 0 !important;
+    padding: 60px !important;
+    width: 100%;
+    /* height: 100%; */
```
Comment (ThaiQ) [inline src/pages/ABCLesson/ABCLesson.css]: remove

File: src/pages/ABCLesson/ABCLesson.css [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    position: relative;
+    padding: 30%;
+    color: black;
+    font-size: x-large;
+    font-style: oblique;
+}
+
+.letter-image{
+    width: 600px;
+    /* max-height: 200px; */
```
Comment (ThaiQ) [inline src/pages/ABCLesson/ABCLesson.css]: remove

File: src/pages/ABCLesson/ABCLesson.css [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    padding-left: 50%;
+}
+
+.word{
+    width: 600px;
+    /* max-height: 200px; */
```
Comment (ThaiQ) [inline src/pages/ABCLesson/ABCLesson.css]: remove

File: src/pages/ABCLesson/ABCLesson.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        };
+        this.next = this.next.bind(this);
+        this.previous = this.previous.bind(this);
+        this.playSound = this.playSound.bind(this);
+
+    }
+
+    playSound = () => {
+
+        console.log("playing..");
```
Comment (ThaiQ) [inline src/pages/ABCLesson/ABCLesson.js]: remove all console logs

File: src/pages/ABCLesson/ABCLesson.js [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+        else {
+            alert("CONGRATS! You have reached the end");
+        }
+        console.log('next button clicked');
+        const nSound = new UIfx(btnSound);
+        nSound.play();
+        console.log(this.state.currentPosition);
+        this.setState({ audioPath: base_URL + '/mp3/' + this.state.audioLetter[this.state.currentPosition] });
+        console.log(this.state.audioPath);
+        //this.setState({ audioVal: this.state.alphabets[this.state.currentPosition].letterSound });
```
Comment (ThaiQ) [inline src/pages/ABCLesson/ABCLesson.js]: remove

File: src/pages/ABCLesson/ABCLesson.js [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+                this.setState({ alphaTick: this.state.alphaTick + 1, currentPosition: this.state.currentPosition });
+            }
+            else {
+                this.setState({ currentPosition: this.state.currentPosition + 1, alphaTick: 0 });
+            }
+        }
+        else {
+            alert("CONGRATS! You have reached the end");
+        }
+        console.log('next button clicked');
```
Comment (ThaiQ) [inline src/pages/ABCLesson/ABCLesson.js]: remove logs

File: src/pages/ABCLesson/ABCLesson.js [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+                        <div className="fields">
+                            <div className="field-block">
+                                {this.state.alphabets[this.state.currentPosition].letter}
+                            </div>
+                        </div>
+                        <div className="buttons">
+                            <a onClick={this.previous} className="button prev">Previous</a>
+                            <a onClick={this.playSound} className="button sound">Play Sound Again
+                        <img className="icon" src="https://i.imgur.com/VoGIU6b.png" /></a>
+                            {/* <a onClick={() => tick.play()} className="button sound">Play Sound Again</a> */}
```
Comment (ThaiQ) [inline src/pages/ABCLesson/ABCLesson.js]: remove

File: src/pages/ABCLesson/ABCLesson.js [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+                    </div>
+                </div>
+            </div>
+
+        );
+
+    }
+}
+
+export default ABCGame;
```
Comment (ThaiQ) [inline src/pages/ABCLesson/ABCLesson.js]: empty lines

File: src/pages/ABCLesson/alphabets.json [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        "wordSound": "Wsound"
+    },
+    {
+        "letter": "Congrats!!!",
+        "image": "https://i.imgur.com/U1C0mTM.png",
+        "word": "",
+        "letterSound": "",
+        "wordSound": ""
+    }
+]
```
Comment (ThaiQ) [inline src/pages/ABCLesson/alphabets.json]: empty line

**Review by ThaiQ — CHANGES_REQUESTED** [review]

**Review by ThaiQ — APPROVED** [review]

**Review by nanarboursal — CHANGES_REQUESTED** [review]
Do the same thing you did for the end at the beginning, so if they click previous before the game starts, you give a message so it doesn't crash. Also, i can't here the sound of the letter being said when i click play sound again

**Review by nanarboursal — APPROVED** [review]
beautiful!!

File: src/pages/ABCLesson/ABCLesson.js [hunk 12]
```diff
@@ -0,0 +1,129 @@
+import React, { Component } from "react";
```
Comment (seemavora) [inline src/pages/ABCLesson/ABCLesson.js:1]: format document on VsCode

**Review by seemavora — APPROVED** [review]

**Pull request merged**